packages/loaders/url/src/index.ts

```diff
@@ -282,6 +287,29 @@ export class UrlLoader implements DocumentLoader<LoadFromUrlOptions> {
     };
   }
 
+  buildWSLegacySubscriber(pointer: string, webSocketImpl: typeof WebSocket, headers: Record<string,string>): Subscriber {
+    const WS_URL = switchProtocols(pointer, {
+      https: 'wss',
+      http: 'ws',
+    });
+    const subscriptionClient = new LegacySubscriptionClient(WS_URL, {
+      connectionParams: () => {
+        return {
+          headers
```

Comment:
Instead of using `headers`, we can have a parameter for `connectionParams` directly. What do you think?

Reply:
Agreed. Also I realized that the same thing applies to the new subscription client as well. Will update and push

Reply:
Done